### PR TITLE
Adds portable configuration possibility

### PR DESCRIPTION
**Pull request by schmic**
As discussed in #116 this adds the possibility to keep the configuration/plugins inside the application folder.

During startup this checks for a subdirectory called "portable" inside the cwd, if found that is used for storing configuration/plugins. Overrides ZAZU_HOME!

### CODE REVIEW

**Comment by schmic** [comment]
Forgot to add `/portable` to .gitignore, please add after merge as well.

**Comment by bayleedev** [comment]
@schmic so what's the workflow? You open up the `.app` folder, create a `portable` folder and then run the app?

**Comment by schmic** [comment]
@blainesch Exactly, if you want to use it portable simple create the "portable" folder.

You can do that in your development as well before running "npm start".

**Comment by bayleedev** [comment]
I'm okay with merging this, but it should be documented somewhere. Maybe in the getting started section? Or maybe the download page since it has instructions for running it locally?

Can you add that to this pr?

**Comment by schmic** [comment]
OK, gonna add the documentation into the PR tomorrow

File: docs/_documentation/configuration.md [hunk 1]
```diff
@@ -21,6 +21,14 @@ basic usage.
 }
 ~~~~
 
+## Portable Mode {#portableMode}
+
+If you want your configuration within your application folder for portability just
+create a `portable` directory inside of it. You can copy your configuration files over
+from your home directory if they already exist.
+
+Portable mode overrides ZAZU_HOME, if defined.
```
Comment (bayleedev) [inline docs/_documentation/configuration.md]: `ZAZU_HOME` isn't documented and was intended for testing purposes. I don't think we should mention it.

File: docs/_documentation/configuration.md [hunk 2]
```diff
@@ -21,6 +21,14 @@ basic usage.
 }
 ~~~~
 
+## Portable Mode {#portableMode}
```
Comment (bayleedev) [inline docs/_documentation/configuration.md]: I think this section should go at the bottom of the page.

File: docs/_documentation/getting-started.md [hunk 3]
```diff
@@ -32,3 +32,5 @@ version. It's located in your home directory under `~/.zazurc.json` (or
 `C:\Users\YOUR_NAME\.zazurc.json` for Windows). Here, you can add some new
 plugins and remove ones you don't want. There is more information in our
 documentation on [Personal Configuration](/documentation/configuration/).
+
+Zazu supports running in portable mode, details are found [here](/documentation/configuration#portableMode)
```
Comment (bayleedev) [inline docs/_documentation/getting-started.md]: details are found on the [configuration page]

**Comment by schmic** [comment]
Docs added, hope that PR is still ok with the merge commit?

File: docs/_documentation/configuration.md (hunk 2 above)
Comment (bayleedev) [reply docs/_documentation/configuration.md]: I had no idea the `{#foo}` thing worked. I was using a js-shim. I should go back and fix that [:

**Review by bayleedev — CHANGES_REQUESTED** [review]

**Comment by bayleedev** [comment]
I'll merge it in manually to get rid of the merge commit. You can rebase it away on this branch if you know how.

**Comment by schmic** [comment]
There you go, no idea how to rebase though but that's fine.

**Review by bayleedev — APPROVED** [review]

**Comment by bayleedev** [comment]
I pushed your code to master:

https://github.com/tinytacoteam/zazu/commits/master

**Pull request closed without merging**